backend/app/repository/github_provider.py

```
@@ -823,3 +823,143 @@ async def get_branch_diff(
+            raise HTTPException(status_code=400, detail="Git token not configured")
+
+        api_base_url = self._get_api_base_url(git_domain)
+
+        try:
+            headers = {
+                "Authorization": f"token {git_token}",
+                "Accept": "application/vnd.github.v3+json",
+            }
+
+            # Get check run details
+            response = requests.get(
```

Ruff (0.14.6)

Probable use of `requests` call without timeout

(S113)